python/tank/bootstrap/manager.py

```diff
@@ -516,7 +516,21 @@ def prepare_engine(self, engine_name, entity):
 
         return path
 
-    def get_pipeline_configurations(self, project):
+    def get_resolved_pipeline_configuration_descriptor(self, project_id=None):
+        """
+        Resolves a pipeline configuration and returns its associated descriptor object.
```

Comment:
The config descriptor's URI is used as part of the unique identifier for a set of engine commands in the sqlite cache that's being build by the websocket server's RPC api. As such, I need a descriptor.